contracts/MarketMint.sol

```diff
@@ -130,91 +128,123 @@ contract Mint is ReentrancyGuard, Pausable {
   mapping (uint256 => NFT) private _idToNft;
   mapping (uint256 => RedemptionToken) private _idToRedemption;
   mapping (address => uint) private _indexToRedemptionToken;
-
+  
   // Event declaration
-  event nftCreated(uint nftId, address creator);
+  event nftClaimed(uint nftId, address creator);
   event Received(address, uint);
 
   /*~~~>
     Restricted access functions for mutable state variables  
   <~~~*/
-  function setRoleAdd(address _role) public hasAdmin returns(bool){
+  function setRoleAdd(address _role) external hasAdmin returns(bool){
     roleAdd = _role;
     return true;
   }
 
   /// @notice
-  /*~~~>
-    Function for re - setting the redemption token details
-  <~~~*/
+    /*~~~>
+      Function for re - setting the redemption token details
+    <~~~*/
   /// @dev
-  /*~~~>
-    uint _redeemAmount: Amount needed to redeem a NFT; 
-    address _contract: address of the redemption token;
-  <~~~*/
-  function resetRedemptionToken(uint _redeemAmount, address _contract) public hasAdmin returns(bool){
+    /*~~~>
+      uint _redeemAmount: Amount needed to redeem a NFT; 
+      address _contract: address of the redemption token;
+    <~~~*/
+  function resetRedemptionToken(uint _redeemAmount, address _contract) external hasAdmin returns(bool){
     uint index = _indexToRedemptionToken[_contract];
     _idToRedemption[index] = RedemptionToken(_redeemAmount, _contract);
     return true;
   }
   
   /// @notice
-  /*~~~> Function for setting new redemption tokens <~~~*/
+    /*~~~> Function for setting new redemption tokens <~~~*/
   /// @dev
-  /*~~~> 
-    uint amount: amount needed to redeem for creating new NFTs;
-    address _toke: address for the token;
-  <~~~*/
+    /*~~~> 
+      uint _redeemAmount: amount needed to redeem for creating new NFTs;
+      address _contract: address for the token;
+    <~~~*/
   /// @return Bool
-  function setNewRedemption(uint amount, address _toke) public hasAdmin returns(bool){
+  function setNewRedemption(uint _redeemAmount, address _contract) external hasAdmin returns(bool){
     _redemptionERC20.increment();
     uint id = _redemptionERC20.current();
-    _idToRedemption[id] = RedemptionToken(amount, _toke);
+    _idToRedemption[id] = RedemptionToken(_redeemAmount, _contract);
     return true;
   }
 
   /// @notice
-  /*~~~>
-  Public interaction function for creating new NFTs by exchanging Tokens
-  <~~~*/
+  //*~~~> For updating the total NFT array with new tokenIds
+    ///@dev Can only be incremented!
+    //*~~~> uint _tokenIds: token Ids to add to the array
+  function setNftTokenIds(uint _amount) external hasDevAdmin returns(bool){
+    for (uint i; i<_amount; i++){
+      totalNfts = totalNfts+1;
+      availableNfts.push(totalNfts);
```

Comment:
Instead of incrementing by +1 _amount times, increment by _amount once. Below or above the for loop

Reply:
the totalNfts will start at 0 by default. When we first enter the initial minted batch, we will enter say 128 i.e.(that's how many are currently prepared) and it will increment and then push that Id onto the array. This way we can just enter the amount minted, and it will increment the totalNfts one by one and push the correct tokenId on. We would need to make the nfts start at one for this, or do a push first and then increment.

Reply:
you're incrementing i in parallel to totalNFTs.